HW04.Rmd

```diff
@@ -1,126 +1,260 @@
+    * Month
+    * Intent
+    * Age and sex
+    * Education
+    * Season
+    * Intent by race
+    * Police-relatedness 
+
+## Read in the data
+
+The gun_deaths.csv is provided in the repository. 
+```{r load-data, message=FALSE, warning=FALSE}
+library(tidyverse)    # load tidyverse packages, including ggplot2
+
+# read in the data
+library(readr)
+gun_death <- read_csv('gun_deaths.csv')
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+library(knitr)
+monthly_death <- gun_death %>%
+    count(month) %>%
+    rename(deaths = n)
+knitr::kable(monthly_death, "markdown",
+             align = 'cc',
+             caption = "Gun deaths per month")
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+library(ggplot2)
+ggplot(monthly_death, aes(x = factor(month), y = deaths - 6000)) +
+    geom_bar(stat = "identity") +
+    scale_x_discrete("Month",
+                     breaks = 1:12,
+                     labels = c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec")) + 
+    scale_y_continuous(name = "Deaths",
+                       limits = c(0,3200),
+                       breaks = c(0, 1000, 2000, 3000),
+                       labels = c("6000", "7000", "8000", "9000"),
+                       expand = c(0,0)) +
+    geom_text(aes(label = deaths), vjust = -1) +
+    labs(title = "Gun deaths per month") +
+    theme(panel.background = element_blank(),
+          axis.line = element_line(),
+          plot.title = element_text(hjust = 0.5))
+    
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+intent_type <- gun_death %>%
+    count(intent, sort = TRUE) %>%
+    rename(deaths = n)
+ggplot(intent_type, aes(x = reorder(intent, -deaths), y = deaths)) +
+    geom_bar(stat = "identity") +
+    scale_y_continuous("Deaths",
+                       expand = c(0,0), 
+                       limits = c(0,70000)) +
+    labs(title = "Number of each type of intent") +
+    xlab("Intent") +
+    geom_text(aes(label = deaths), vjust = -1) +
+    theme(panel.background = element_blank(),
+          axis.line = element_line(),
+          plot.title = element_text(hjust = 0.5))
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+age_sex <- gun_death %>%
+    select(age, sex)
+age_sex = subset(age_sex, age != "Not Available")
```

Comment:
I thought this was an interesting way to go about this. It didn't occur to me that one can also use subset to remove NA age. I just use filter, this is good to know!